akka-remote/src/main/scala/akka/remote/transport/netty/NettyTransport.scala

```diff
@@ -238,16 +238,32 @@ private[transport] object NettyTransport {
   val uniqueIdCounter = new AtomicInteger(0)
 
   def addressFromSocketAddress(addr: SocketAddress, schemeIdentifier: String, systemName: String,
-                               hostName: Option[String], port: Option[Int]): Option[Address] = addr match {
-    case sa: InetSocketAddress ⇒ Some(Address(schemeIdentifier, systemName,
-      hostName.getOrElse(sa.getAddress.getHostAddress), port.getOrElse(sa.getPort))) // perhaps use getHostString in jdk 1.7
-    case _ ⇒ None
+                               hostName: Option[String], port: Option[Int]): Option[Address] = {
+
+    addr match {
+      case sa: InetSocketAddress ⇒
+        Some(Address(
+          schemeIdentifier,
+          systemName,
+          ipv6SafeHostname(hostName.getOrElse(sa.getHostString)),
```

Comment:
we're JDK8 now here, so please `getHostString` :-) as hinted in comment `// perhaps use getHostString in jdk 1.7`